src/Aspire.Hosting.Qdrant/QdrantBuilderExtensions.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.Qdrant;
+using Aspire.Hosting.Utils;
+
+namespace Aspire.Hosting;
+
+/// <summary>
+/// Provides extension methods for adding Qdrant resources to the application model.
+/// </summary>
+public static class QdrantBuilderExtensions
+{
+    private const int QdrantPortGrpc = 6334;
+    private const int QdrantPortHttp = 6333;
+    private const string ApiKeyEnvVarName = "QDRANT__SERVICE__API_KEY";
+    private const string EnableStaticContentEnvVarName = "QDRANT__SERVICE__ENABLE_STATIC_CONTENT";
+
+    /// <summary>
+    /// Adds a Qdrant resource to the application. A container is used for local development.  
+    /// </summary>
+    /// <remarks>
+    /// This version the package defaults to the v1.8.3 tag of the qdrant/qdrant container image.
+    /// The .NET client library uses the gRPC port by default to communicate and this resource exposes that endpoint.
+    /// </remarks>
+    /// <param name="builder">The <see cref="IDistributedApplicationBuilder"/>.</param>
+    /// <param name="name">The name of the resource. This name will be used as the connection string name when referenced in a dependency</param>
+    /// <param name="apiKey">The parameter used to provide the API Key for the Qdrant resource. If <see langword="null"/> a random key will be generated as {name}-Key.</param>
+    /// <param name="port">The host port of Qdrant database.</param>
+    /// <returns>A reference to the <see cref="IResourceBuilder{QdrantServerResource}"/>.</returns>
+    public static IResourceBuilder<QdrantServerResource> AddQdrant(this IDistributedApplicationBuilder builder,
+        string name,
+        IResourceBuilder<ParameterResource>? apiKey = null,
+        int? port = null)
+    {
+        var apiKeyParameter = apiKey?.Resource ??
+            ParameterResourceBuilderExtensions.CreateDefaultPasswordParameter(builder, $"{name}-Key", special: false);
+        var qdrant = new QdrantServerResource(name, apiKeyParameter);
+        return builder.AddResource(qdrant)
+            .WithImage(QdrantContainerImageTags.Image, QdrantContainerImageTags.Tag)
+            .WithHttpEndpoint(port: port, targetPort: QdrantPortGrpc, name: QdrantServerResource.PrimaryEndpointName)
```

Comment:
Using the same port for grpc and http is not correct. Specifying a port will clash.

Reply:
Resolved this with https://github.com/dotnet/aspire/pull/3422